ddl/partition.go

```diff
@@ -457,20 +457,27 @@ func isRangePartitionColUnsignedBigint(cols []*table.Column, pi *model.Partition
 	return false
 }
 
-// truncateTableByReassignPartitionIDs reassign a new partition ids.
-func truncateTableByReassignPartitionIDs(job *model.Job, t *meta.Meta, tblInfo *model.TableInfo) error {
+// truncateTableByReassignPartitionIDs reassigns new partition ids.
+func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo) error {
 	newDefs := make([]model.PartitionDefinition, 0, len(tblInfo.Partition.Definitions))
 	for _, def := range tblInfo.Partition.Definitions {
 		pid, err := t.GenGlobalID()
 		if err != nil {
-			job.State = model.JobStateCancelled
 			return errors.Trace(err)
 		}
-		newDef := model.PartitionDefinition{
-			ID:       pid,
-			Name:     def.Name,
-			LessThan: def.LessThan,
-			Comment:  def.Comment,
+
```

Comment:
@crazycs520 What happens if I make a mistake in the process of truncate table and then we change part of the partition id?

Reply:
@tiancaiamao PTAL

Reply:
I agree with @crazycs520,  partial changing of the tableInfo will not happen, truncate table will create a new table, and all these changes are in one transaction.

Reply:
If there is an error, there may be an intermediate state in memory.

Reply:
@ciscoxll   `tblInfo` in this function is not get from infoschema, it is get from tikv and unmarshal, so it won't affect the infoschema in memory.

Reply:
if the `tblInfo *model.TableInfo` is a reference from current schema meta, it would be a problem.
A function should make less assumption of the caller.
Maybe add test `GenGlobalID` return error would make our code more robust.

Reply:
Current code is more safe the the proposed one @crazycs520

Reply:
If you wanna prove current code is more safe than the proposed one， Please show me the test code to prove the proposed code is not safe.

Reply:
I don't want to argue with you or, prove I am right, everybody may be biased.
I just wanna the code to be more stable.

Reply:
@crazycs520 Wait until we're modifying the code when implementing the list type partition.